### PR TITLE
Converting randomized testing to create a separate unitTest task instead of replacing the builtin test task

### LINKED ISSUE #33343: [Build] Gradle 5.0 deprecation: should not remove tasks
Gradle deprecated removing a task if it's added as a dependency. 
```
Do not remove a Task instance from a task dependency set when it contains a Provider to the Task instance. This behaviour has been deprecated and is scheduled to be removed in Gradle 5.0.
	at org.gradle.api.internal.tasks.DefaultTaskDependency$TaskDependencySet.remove(DefaultTaskDependency.java:245)
	at java_util_Set$remove$0.call(Unknown Source)
	at com.carrotsearch.gradle.junit4.RandomizedTestingPlugin.replaceTestTask(RandomizedTestingPlugin.groovy:98)
	at com.carrotsearch.gradle.junit4.RandomizedTestingPlugin$replaceTestTask$0.callStatic(Unknown Source)
	at com.carrotsearch.gradle.junit4.RandomizedTestingPlugin.apply(RandomizedTestingPlugin.groovy:22)
	at com.carrotsearch.gradle.junit4.RandomizedTestingPlugin.apply(RandomizedTestingPlugin.groovy)
[...]
```

"Provider to the Task instance" is part of the delayed task ( a.k.a lazy task ) improvements added in 4.8, documented and refined in 4.9, and in this case it's not something we can control as the initial `test` task is represented as a provider, we have no control to change that, we also don't have control to prevent adding it as a dependency. 

This will effectively prevent us from replacing the `test` task in Gradle 5.0.
Some options that come to mind:
 - disable `test` and create a new `randomizedTest` task instead 
 - use the Gradle test runner. This will require us to configure the security manager as described in #31496.  We will not have randomization of class order unless gradle/gradle#5760 is addressed . We will still have randomization of method order. 
 - we can probably start hacking around it, like excluding all tests and adding a task action to run the randomized testing ant runner, but we don't want to go there.

### PR DESCRIPTION
**Pull request by alpar-t**
Gradle 5.0 deprecates removing dependencies from tasks and replacing tasks with a different implementation.  
This means that replacing the `test` task will be impossible for Gradle 6.0. 

This PR introduced a `unitTest` task instead. The original `test` is now disabled and depended on  `unitTest`.  This means that running `./gradlew test` will work as before.
With this change we will be able to continue upgrading Gradle regardless of #31496. 
I also think that the `test` tasks name is too broad anyhow and probably only kept for historical reasons and because different projects split the tests differently.
With this PR `test` will be much more like `check` just a way to group other tasks together. 

Closes #33343

### CODE REVIEW

**Comment by rjernst** [comment]
This PR seems to have some unmerged other PRs in it, causing the diff to be around +35k lines...can you please re-ping for review when that is resolved?

**Comment by atorok** [comment]
@rjernst Not sure what you mean, the diff looks normal to me `+85 −122 ` ( there are a lot f one line changes doing `s/test/unitTest/` so 67 files ). Maybe there was some glitch ?

File: buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/RandomizedTestingPlugin.groovy [hunk 1]
```diff
@@ -57,35 +41,16 @@ class RandomizedTestingPlugin implements Plugin<Project> {
...
-            // no test task, ok, user will use testing task on their own
-            return
+    static void createUnitTestTask(TaskContainer tasks) {
+        // only create a unitTest task if the `test` task exists as some project don't make use of it.
+        tasks.matching { it.name == "test" }.all {
+            // We don't want to run any tests with the Gradle test runner since we add our own randomized runner
+            it.enabled = false
+            RandomizedTestingTask unitTest = tasks.create('unitTest', RandomizedTestingTask)
+            unitTest.description = 'Runs unit tests with the randomized testing framework'
+            unitTest.include("**/*Tests.class")
```
Comment (rjernst) [inline buildSrc/src/main/groovy/com/carrotsearch/gradle/junit4/RandomizedTestingPlugin.groovy]: I think which tests are run should stay outside of RandomizedTesting code. It is still my intention to get this back into the randomizedtesting project, in which case the test naming patterns should not be assumed here.

**Review by rjernst — APPROVED** [review]
I'm not sure why github showed me a such a weird diff before; it looks fine now.

This looks fine to me. In addition to my one comment, I think the PR title needs to be changed, as it is only a side effect of the actual change: converting randomized testing to create a separate unitTest task instead of replacing the builtin test task.

**Pull request merged**